File: Workbench/src/main/java/me/zhang/workbench/binding/DataBindingActivity.java
package me.zhang.workbench.binding;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.ViewGroup;

import me.zhang.workbench.databinding.ActivityDbBinding;

public class DataBindingActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

//        ActivityDbBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_db);
        ActivityDbBinding binding = ActivityDbBinding.inflate(getLayoutInflater(),
                (ViewGroup) findViewById(android.R.id.content), true);
        binding.setUser(new User("Test", "User"));
    }
}

File: app/src/main/java/me/zhang/laboratory/ui/databinding/DataBindingActivity.kt
package me.zhang.laboratory.ui.databinding

import android.os.Bundle
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import me.zhang.laboratory.R
import me.zhang.laboratory.databinding.ActivityBinding

class DataBindingActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding =
            DataBindingUtil.setContentView<ActivityBinding>(this, R.layout.activity_binding)
        val userViewModel by viewModels<UserViewModel>()
        userViewModel.setUser(name = "Zhang", age = 31, arrayListOf("Coding", "Cooking", "Surfing"))
        userViewModel.setAccount(username = "zhang", password = "123456")

        binding.viewModel = userViewModel
        binding.handler = object : Handler {
            override fun layoutChanged() {
                Toast.makeText(this@DataBindingActivity, "layout changed", Toast.LENGTH_SHORT)
                    .show()
            }
        }
        binding.lifecycleOwner = this
        binding.executePendingBindings()
    }
}

interface Handler {
    fun layoutChanged()
}